src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1259,7 +1237,7 @@ internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavio
                         // to prevents a dependency on the thread pool (which would also trigger deadlocks).
                         // The WriteBuffer notifies this command when the first buffer flush occurs, so that the
                         // send functions can switch to the special async mode when needed.
-                        var sendTask = NonMultiplexingWriteWrapper(connector, async, finalCt);
+                        var sendTask = NonMultiplexingWriteWrapper(connector, async, CancellationToken.None);
```

Comment:
> If we really end up going down this path, where buffers exclusively use their own CTS wrappers and never receive a token from the outside, we'd remove all this again :)

We're definitely not removing anything. `NpgsqlConnection.WaitAsync` passes the cancellation token to the buffer.

Reply:
Why not have the WaitAsync token just trigger the buffer's CTS, more or less like how we do elsewhere (but without triggering PostgreSQL cancellation of course)?

Reply:
Because then we have to catch the original exception and throw the `OperationCancelledException`.

Reply:
We can definitely leave this for later, but it would be really weird if all our types (and all type handlers) accept a cancellation token that's almost always None just in order to support WaitAsync... Let's revisit after the rest is done.

Reply:
While I'm definitely not against this idea, I do think it requires a thorough evaluation. A lot of work has been done, and it would be a shame to just throw it away.

Reply:
While I'm definitely not against this idea, I do think it requires a thorough evaluation. A lot of work has been done, and it would be a shame to just throw it away.

Reply:
I know, and I don't underestimate how much effort went into adding all those cancellation tokens. But at the end of the day, the question is whether the design we end up with needs them or not...

Let's defer the question until we're happy with everything else, then we can take a look and make a decision.